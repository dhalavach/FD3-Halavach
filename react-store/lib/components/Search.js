"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Search;
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function Search(props) {
  var searchQuery = props.searchQuery,
    handleSearchProducts = props.handleSearchProducts;
  return <div>
      Search{' '}
      <input name='search' id='search-box' type='search' placeholder='search products' value={searchQuery} onChange={handleSearchProducts} />
    </div>;
}
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJuYW1lcyI6WyJfcmVhY3QiLCJfaW50ZXJvcFJlcXVpcmVEZWZhdWx0IiwicmVxdWlyZSIsIm9iaiIsIl9fZXNNb2R1bGUiLCJkZWZhdWx0IiwiU2VhcmNoIiwicHJvcHMiLCJzZWFyY2hRdWVyeSIsImhhbmRsZVNlYXJjaFByb2R1Y3RzIl0sInNvdXJjZXMiOlsiLi4vLi4vc3JjL2NvbXBvbmVudHMvU2VhcmNoLnRzeCJdLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgUmVhY3QsIHsgRnVuY3Rpb25Db21wb25lbnQgfSBmcm9tICdyZWFjdCc7XHJcblxyXG5leHBvcnQgZGVmYXVsdCBmdW5jdGlvbiBTZWFyY2gocHJvcHM6IGFueSkge1xyXG4gIGNvbnN0IHsgc2VhcmNoUXVlcnksIGhhbmRsZVNlYXJjaFByb2R1Y3RzIH0gPSBwcm9wcztcclxuICByZXR1cm4gKFxyXG4gICAgPGRpdj5cclxuICAgICAgU2VhcmNoeycgJ31cclxuICAgICAgPGlucHV0XHJcbiAgICAgICAgbmFtZT0nc2VhcmNoJ1xyXG4gICAgICAgIGlkPSdzZWFyY2gtYm94J1xyXG4gICAgICAgIHR5cGU9J3NlYXJjaCdcclxuICAgICAgICBwbGFjZWhvbGRlcj0nc2VhcmNoIHByb2R1Y3RzJ1xyXG4gICAgICAgIHZhbHVlPXtzZWFyY2hRdWVyeX1cclxuICAgICAgICBvbkNoYW5nZT17aGFuZGxlU2VhcmNoUHJvZHVjdHN9XHJcbiAgICAgIC8+XHJcbiAgICA8L2Rpdj5cclxuICApO1xyXG59XHJcbiJdLCJtYXBwaW5ncyI6Ijs7Ozs7O0FBQUEsSUFBQUEsTUFBQSxHQUFBQyxzQkFBQSxDQUFBQyxPQUFBO0FBQWlELFNBQUFELHVCQUFBRSxHQUFBLFdBQUFBLEdBQUEsSUFBQUEsR0FBQSxDQUFBQyxVQUFBLEdBQUFELEdBQUEsS0FBQUUsT0FBQSxFQUFBRixHQUFBO0FBRWxDLFNBQVNHLE1BQU1BLENBQUNDLEtBQVUsRUFBRTtFQUN6QyxJQUFRQyxXQUFXLEdBQTJCRCxLQUFLLENBQTNDQyxXQUFXO0lBQUVDLG9CQUFvQixHQUFLRixLQUFLLENBQTlCRSxvQkFBb0I7RUFDekMsT0FDRSxDQUFDLEdBQUcsQ0FBQztBQUNULFlBQVksQ0FBQyxHQUFHLENBQUM7QUFDakIsTUFBTSxDQUFDLEtBQUssQ0FDSixJQUFJLENBQUMsUUFBUSxDQUNiLEVBQUUsQ0FBQyxZQUFZLENBQ2YsSUFBSSxDQUFDLFFBQVEsQ0FDYixXQUFXLENBQUMsaUJBQWlCLENBQzdCLEtBQUssQ0FBQyxDQUFDRCxXQUFXLENBQUMsQ0FDbkIsUUFBUSxDQUFDLENBQUNDLG9CQUFvQixDQUFDLEdBQy9CO0FBQ1IsSUFBSSxFQUFFLEdBQUcsQ0FBQztBQUVWIn0=