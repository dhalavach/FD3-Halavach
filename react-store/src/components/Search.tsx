import { ChangeEvent } from "react";
import useSearchParamsState from "../hooks/useSearchParamsState";

export default function Search() {
  const [searchQueryParam, setSearchQueryParam] = useSearchParamsState(
    "searchQueryParam",
    ""
  );
  return (
    <div>
      Search{" "}
      <input
        name="search"
        id="search-box"
        type="search"
        placeholder="search products"
        value={searchQueryParam}
        data-testid="search__input-field"
        onChange={(e: ChangeEvent<HTMLInputElement>) =>
          setSearchQueryParam(e.target.value)
        }
      />
    </div>
  );
}
