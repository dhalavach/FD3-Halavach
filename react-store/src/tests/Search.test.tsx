import { cleanup, fireEvent, render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";

import Search from "../components/Search";
import { BrowserRouter as Router } from "react-router-dom";
import store from "../components/Store";
import { Provider } from "react-redux";

afterEach(cleanup);

test("Renders the component", () => {
  expect(
    render(
      <Provider store={store}>
        <Router>
          <Search />
        </Router>
      </Provider>
    )
  ).toBeTruthy();
});

test("search input field changes when input is entered", async () => {
  render(
    <Provider store={store}>
      <Router>
        <Search />
      </Router>
    </Provider>
  );
  const searchInput = screen.getByTestId("search__input-field");
  fireEvent.change(searchInput, { target: { value: "qwerty" } });
  const searchInputValue = screen.getByTestId(
    "search__input-field"
  ) as HTMLInputElement;
  expect(searchInputValue.value).toBe("qwerty");
});

test("component matches the snapshot", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <Search />
        </Router>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
