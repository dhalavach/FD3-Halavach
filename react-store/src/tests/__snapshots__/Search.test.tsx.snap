// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component matches the snapshot 1`] = `
<div>
  Search
   
  <input
    data-testid="search__input-field"
    id="search-box"
    name="search"
    onChange={[Function]}
    placeholder="search products"
    type="search"
    value="qwerty"
  />
</div>
`;
